### PR TITLE
Remove need for extra conditional, and anonymous functions

### PR DESCRIPTION
**Pull request by nixpulvis**
This is two commits:

1. Removes the extra lambdas, making use of `bind(this)`. I have no *real* idea about performance but I can only imagine this helping, plus it's a bit cleaner.
2. Remove extra conditional for `this.lastTime`. This will help with performance by exactly one conditional :stuck_out_tongue_winking_eye: and makes the code better.

Feel free to ignore this PR, I'm just sharing my findings with you.

### CODE REVIEW

**Comment by nixpulvis** [comment]
Possibly interesting, though it seems there's a lot of confusion and mixed messages out there (lolz JS):

- https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1
- https://github.com/facebook/react/issues/9851
- https://stackoverflow.com/questions/42117911/lambda-functions-vs-bind-memory-and-performance

File: js/renderer.js [hunk 1]
```diff
@@ -115,8 +115,8 @@ class Renderer {
         this.initBullets(data[2]);
         this.initShips(data[3]);
 
-        this.lastTime = -1
-        requestAnimationFrame((ts) => this.renderLoop(ts));
+        this.lastTime = performance.now();
```
Comment (ryan-brown) [inline js/renderer.js]: This does not work. If you console.log(this.lastTime) vs console.log(ts) in the first iteration of renderLoop, you will see that this.lastTime is over 100ms later than the given ts. They must be getting their timestamp from different sources or at different times.

**Review by ryan-brown — CHANGES_REQUESTED** [review]

File: js/renderer.js (hunk 1 above)
Comment (nixpulvis) [reply js/renderer.js]: First of all, with the way you've implemented your moving average, the first 10 frames are going to be a progressively better estimate, but the first 2 or 3 are still quite wrong. That's fine as far as I'm concerned, but could probably be better with something that takes into account the number of elements actually added to the list.

But now that I play with this a little more, I understand better what you are saying. Basically there is a [bug in how this works](https://stackoverflow.com/questions/38360250/requestanimationframe-now-vs-performance-now-time-discrepancy). Also worth noting, Firefox broke the spec for a lot of this to mitigate spectre, while other browsers did not. For example `performance.now()` in FF is always truncated, while in Chrome it's got decimal points.

Fuck I hate web development.

**Comment by nixpulvis** [comment]
@ryan-brown updated to still avoid dummy `-1` value, but correctly deal with the fact that `performance.now` is a shit show.

**Pull request closed without merging**